networking/gcp/vpc/outputs.tf

```
@@ -0,0 +1,88 @@
+output "name" {
+  description = "The name of the VPC"
+  value       = google_compute_network.vpc.name
+}
+
+output "id" {
+  description = "The ID of the VPC"
+  value       = google_compute_network.vpc
+}
+
+output "gateway_ipv4" {
+  description = "The gateway address for default routing out of the network. This value is selected by GCP"
+  value       = google_compute_network.vpc.gateway_ipv4
+}
+
+output "self_link" {
+  description = "The URI of the created resource"
+  value       = google_compute_network.vpc.self_link
+}
+
+output "private_subnet_ids" {
+  description = "List of private subnet IDs"
+  value       = { for subnet, _ in local.private_subnets : subnet => google_compute_subnetwork.subnets[subnet].id }
+}
+
+output "private_subnet_cidr_blocks" {
+  description = "List of private subnet CIDR blocks"
+  value       = { for subnet, _ in local.private_subnets : subnet => google_compute_subnetwork.subnets[subnet].ip_cidr_range }
+}
+
+output "private_subnet_self_links" {
+  description = "List of private subnet self links"
+  value       = { for subnet, _ in local.private_subnets : subnet => google_compute_subnetwork.subnets[subnet].self_link }
+}
+
+output "private_subnet_regions" {
+  description = "List of private subnet regions"
+  value       = { for subnet, _ in local.private_subnets : subnet => google_compute_subnetwork.subnets[subnet].region }
+}
```

I would make it a single output:

```suggestion
output "private_subnet" {
  description = "List of private subnet IDs"
  value = { for subnet, _ in local.private_subnets : subnet => {
    id         = google_compute_subnetwork.subnets[subnet].id
    cidr_range = google_compute_subnetwork.subnets[subnet].ip_cidr_range
    link       = google_compute_subnetwork.subnets[subnet].self_link
    region     = google_compute_subnetwork.subnets[subnet].region
  }
}
```

(and the same with public subnets)

---

As far as possible, we prefer to have finer outputs at module level, so that documentation on each output is easier to read and to understand.